src/main/java/org/broadinstitute/hellbender/engine/FeatureInput.java

```diff
@@ -70,6 +87,123 @@
      */
     public static final String FEATURE_ARGUMENT_KEY_VALUE_SEPARATOR = "=";
 
+    public static boolean isGenomeDBPath(String filePath){
+        return filePath.startsWith(GENOMIC_DB_URI_SCHEME);
+    }
+
+    @SuppressWarnings("unchecked")
+    private FeatureCodec<T, ?> getCodec() {
+        if( codec != null){
+            return codec;
+        }
+
+        if( featureType == null){
+            throw new GATKException("setFeatureType must be called before calling getCodec");
+        }
+        codec = (FeatureCodec<T, ?>) FeatureManager.getCodecForFile(new File(getFeaturePath()), featureType);
+
+        return codec;
+    }
+
+    @SuppressWarnings("unchecked")
+    public FeatureReader<T> getFeatureReader() {
```

Comment:
One thing I don't like about having `getFeatureReader()` in the `FeatureInput` is that it makes it impossible to enforce the constraint imposed by the Tribble API that there be no more than one open iteration per reader. When the reader was encapsulated within `FeatureDataSource`, this was easy to enforce.